wasp-app-runner/src/build/client.ts

```diff
@@ -0,0 +1,82 @@
+import * as path from "path";
+import * as fs from "fs";
+
+import { spawnWithLog } from "../process.js";
+import { log } from "../logging.js";
+import { EnvVars } from "../types.js";
+import { parse } from "dotenv";
+import { doesFileExits } from "../files.js";
+import type { PathToApp } from "../args.js";
+
+export async function buildAndStartClientApp({
+  pathToApp,
+}: {
+  pathToApp: PathToApp;
+}): Promise<void> {
+  const { exitCode: buildExitCode } = await buildClientApp({ pathToApp });
+  if (buildExitCode !== 0) {
+    log("client-build-app", "error", `Failed to build client app.`);
+    process.exit(1);
+  }
+
+  // This starts a long running process, so we don't await it.
+  startClientApp({ pathToApp }).then(({ exitCode }) => {
+    if (exitCode !== 0) {
+      log("client-start-app", "error", `Failed to start client app.`);
+      process.exit(1);
+    }
+  });
+}
+
+async function buildClientApp({
+  pathToApp,
+}: {
+  pathToApp: PathToApp;
+}): Promise<{ exitCode: number | null }> {
+  const defaultRequiredEnv = {
+    REACT_APP_API_URL: "http://localhost:3001",
+  };
+  const devEnv = await getDevEnvVars({ pathToApp });
+
+  const clientBuildEnv = {
+    ...defaultRequiredEnv,
+    ...devEnv,
+  };
+
+  return spawnWithLog({
+    name: "client-build-app",
+    cmd: "npm",
+    args: ["run", "build"],
+    cwd: path.join(pathToApp, ".wasp/build/web-app"),
+    extraEnv: clientBuildEnv,
+  });
+}
+
+async function startClientApp({
+  pathToApp,
+}: {
+  pathToApp: PathToApp;
+}): Promise<{
+  exitCode: number | null;
+}> {
+  return spawnWithLog({
+    name: "client-start-app",
+    cmd: "npx",
+    args: ["serve", "--single", "-p", "3000"],
+    cwd: path.join(pathToApp, ".wasp/build/web-app/build"),
```

Comment:
We're 100% going to forget to change this (and other hardcoded strings) at one point, but I see no better solution :smile: 


One thing we could improve at least: `.wasp/build/web-app` is mentioned in two places. I recommend extracting that into a constant (inside this file).

Reply:
I'd put them at the bottom of the file, but not that important. Resolve when happy.